crates/sargon/src/system/sargon_os/entities_linked_to_factor_source/sargon_os_entities_linked_to_factor_source.rs

```diff
@@ -0,0 +1,133 @@
+use crate::prelude::*;
+
+impl SargonOS {
+    /// Returns the entities linked to a given `FactorSource`, either on the current `Profile` or a specific one.
+    pub async fn entities_linked_to_factor_source(
+        &self,
+        factor_source: FactorSource,
+        profile_to_check: ProfileToCheck,
+    ) -> Result<EntitiesLinkedToFactorSource> {
+        let accessibility = self.accessibility(factor_source.clone()).await?;
```

Comment:
Same review comment as in some PRs ago... :) this should not be on `SargonOS` :) `SargonOS` ought to only "forward" calls to `Profile` (via `ProfileHolder`.). since you here support `ProfileToCheck::Specific` so maybe something like?:


```rust
impl SargonOS {
    /// Returns the entities linked to a given `FactorSource`, either on the current `Profile` or a specific one.
    pub async fn entities_linked_to_factor_source(
        &self,
        factor_source: FactorSource,
        profile_to_check: ProfileToCheck,
    ) -> Result<EntitiesLinkedToFactorSource> {

        let existence = self.claimed_and_actual_existence_of_factor_source(factor_source.clone()).await?;

        match profile_to_check {
            ProfileToCheck::Specific(profile) => profile.entities_linked_to_factor_source(factor_source, existence),
            ProfileToCheck::Current => self.profile().entities_linked_to_factor_source(factor_source, existence),
        }
    }
}
```